### PR TITLE
feat: make win.setAspectRatio() work on Windows

### PR DESCRIPTION
**Pull request by zcbenz**
#### Description of Change

Close https://github.com/electron/electron/issues/8036.

This PR makes `win.setAspectRatio` work on Windows. It didn't work previously because the Windows implementation requires a non-zero maximum size, while we provided (0, 0) when maximum size is not set.

There is also a patch added to allow setting aspect ratio to 0 on Windows.

I'm unable to write a test for this because the aspect ratio is not respected when window is resized programmingly with APIs like `win.setSize`, it is also why we did not have a test for it at the first place.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Make `win.setAspectRatio()` work on Windows.

### CODE REVIEW

**Review by samuelmaddock — APPROVED** [review]
API LGTM. A test might be nice for the aspect ratio of 0 edge case to ensure the patch continues to work.

File: patches/chromium/enable_reset_aspect_ratio.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+diff --git a/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc b/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc
+index 3b132d4d7618e11b851b315e82fa67442c852fb3..6cee855828fc8069b7cd7d429dcda5dc9bd7ecf8 100644
+--- a/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc
++++ b/ui/views/widget/desktop_aura/desktop_window_tree_host_win.cc
+@@ -484,7 +484,6 @@ void DesktopWindowTreeHostWin::SetOpacity(float opacity) {
+ }
+ 
+ void DesktopWindowTreeHostWin::SetAspectRatio(const gfx::SizeF& aspect_ratio) {
+-  DCHECK(!aspect_ratio.IsEmpty());
```
Comment (jkleinsc) [inline patches/chromium/enable_reset_aspect_ratio.patch:16]: @zcbenz why are we removing this DCHECK?  if aspect_ratio.IsEmpty() then the next line won't work will it?

Comment (zcbenz) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: `aspect_ratio.IsEmpty()` could also be true when it is something like `(0, 100)`.

Comment (codebytere) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: @zcbenz true, but i think this will crash on divide-by-zero if we don't at least ensure that height > 0, won't it?

Comment (zcbenz) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: Yeah it will crash if height == 0, but I think it should be fine crashing here? If someone pass a 0 height then the code will crash anyway with or without the DCHECK. The purpose of the DCHECK is prevent passing 0 to `SetAspectRatio`, instead of prevent the divide-by-zero crash.

Also we won't trigger the divide-by-zero crash in Electron since we are always passing `(ratio, 100)` to this API.

Comment (jkleinsc) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: @zcbenz I guess to ask the question another way, it is necessary to remove this DCHECK?  What problem does it solve to do so?

Comment (zcbenz) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: The Electron `win.setAspectRatio` API accepts `0` to cancel the aspect ratio, which internally passes `(0, 100)` to `DesktopWindowTreeHostWin::SetAspectRatio`, and it would cause an assertion with this DCHECK since `(0, 100)` is considered as an empty size.

Comment (jkleinsc) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: @zcbenz couldn't we change the DCHECK to 
`DCHECK(!aspect_ratio.height());` to avoid division by zero?

Comment (codebytere) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: I think i'd prefer that as well, since it's a clearer error

Comment (zcbenz) [reply patches/chromium/enable_reset_aspect_ratio.patch:16]: @jkleinsc @codebytere Sure, I have changed the DCHECK to validate the height.

**Review by jkleinsc — APPROVED** [review]

**Review by codebytere — APPROVED** [review]

**Comment by samuelmaddock** [comment]
@electron/wg-releases @zcbenz I'm interested in seeing this work in 11.x.y. Any chance we could get it backported?
